src/collective/upload/viewlets.py

```diff
@@ -24,15 +25,39 @@ def enabled(self):
 
 class WidgetViewlet(ViewletBase):
 
+    @property
     def enabled(self):
         """Show the viewlet just in folder_contents view if enabled."""
         if not IFolderContentsView.providedBy(self.view):
             return False
 
         try:
-            return api.portal.get_registry_record(
-                interface=IUploadSettings, name='show_widget')
+            record = dict(interface=IUploadSettings, name='show_widget')
+            return api.portal.get_registry_record(**record)
         except (InvalidParameterError, KeyError):
             # avoid breaking page rendering if record is not present
             # this could happen on upgrades or accidental deletions
             return False
+
+    @property
+    def extensions(self):
+        record = dict(interface=IUploadSettings, name='upload_extensions')
```

Comment:
instead of calling `api.portal.get_registry_record` use the proxy record:

```python
registry = getUtility(IRegistry)
settings = registry.forInterface(IUploadSettings)
```